### PR TITLE
Insert crossbordering gas pipelines with Germany for eGon100RE

### LINKED ISSUE #881: Cross bordering pipelines with Germany in eGon100RE
Cross bordering pipelines (for H2 and CH4 transport) with Germany in eGon100RE are missing.

### PR DESCRIPTION
**Pull request by AmeliaNadal**
Fixes #881  .

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted.
- [x] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

**Review by IlkaCu — APPROVED** [review]

**Comment by IlkaCu** [comment]
Does this only affect the eGon100RE scenario? As we want to focus on finishing the eGon2035 scenario first, it might be a good idea to not merge any eGon100RE stuff into dev, to avoid any unpredicted side effects.

**Comment by AmeliaNadal** [comment]
The main changes are related with eGon100RE, but there are several structural changes in the code that are concerning both scenarios. I just saw your message after merging the PR, I'm sorry.

**Pull request merged**
